Add KB article explaining default rule modes and how to change them (#1830)
semgrep/semgrep-docs: docs/kb/rules/ruleset-default-mode.md

---

> description: Adjust the default mode for a whole ruleset.

Reviewer: ```suggestion
description: Change the default mode for a ruleset.
```

tags:
  - Rules
  - Semgrep Registry
  - Semgrep Code
append_help_link: true
---

# Why do new rules keep appearing in Comment or Block mode?

> Semgrep AppSec Platform [Policies](/docs/semgrep-code/policies) can contain both individual rules and **rulesets**, which are curated groups of rules recommended for particular purposes. All organizations in the AppSec Platform start with two rulesets: the `default` ruleset and the `comment` ruleset. These rulesets are curated to be good starter packs for security teams to review, and for developers to see comments about, respectively.

Reviewer: ```suggestion
Semgrep AppSec Platform [Policies](/docs/semgrep-code/policies) can contain both individual rules and **rulesets**, which are curated groups of rules recommended for particular purposes. All organizations start with two rulesets: the `default` ruleset, which is a good starter pack for security teams, and the `comment` ruleset, which is a good starter pack for developers.
```

I was a bit confused about what "These rulesets are curated to be good starter packs for security teams to review, and for developers to see comments about, respectively" meant  -- do you mean the `default` ruleset is good for security teams to start reviewing findings?

Author: yeah, your version is better - that was some bad writing on my part, sorry!

> As Semgrep adds new rules to improve coverage, some of these rules are also added to rulesets. If you have a ruleset added to your organization's policies, any new rules will be added to the ruleset automatically, and become part of your policies also.

Reviewer: ```suggestion
As Semgrep adds new rules to improve coverage, some of these rules are also added to rulesets. If you add a ruleset to your organization's policies, any new rules added to the ruleset automatically become a part of your policies as well.
```

> The `default` and `comment` rulesets are initially added in Monitor mode, which is primarily intended for security teams to review. You can also [add new rulesets to your policies](/docs/semgrep-code/policies#add-rulesets-to-your-policies-from-the-registry) from the Semgrep Registry.

Reviewer: ```suggestion
The `default` and `comment` rulesets are initially added in **Monitor** mode, where the findings generated by the rules are primarily intended for security teams to review. You can also [add new rulesets to your policies](/docs/semgrep-code/policies#add-rulesets-to-your-policies-from-the-registry) from the Semgrep Registry.
```
I don't know if I captured the gist of what you're saying, but I think I'm trying to clarify whether the security team is reviewing the rules themselves or the findings they generate

> When you add a ruleset through the registry, you can add it in any policy mode: Monitor, Comment, or Block. The mode you choose will determine the mode for future rules that are added to that ruleset.

Reviewer: ```suggestion
When you add a ruleset through the registry, you can add it in any policy mode: **Monitor**, **Comment**, or **Block**. The mode you choose will determine the mode for future rules that are added to that ruleset.
```

> Even if you later move some or all rules from a ruleset to a different mode, the default mode for the ruleset remains the same as it was when it was originally added, and it might be somewhat surprising when new rules are added in the original mode.

Reviewer: ```suggestion
Even if you later change some or all rules from a ruleset to a different mode, the default mode for the ruleset does not change. Therefore, when you add new rules to the ruleset, they are added in the original mode.
```

> ## Changing the default mode for a ruleset

Reviewer: ```suggestion
## Change the default mode for a ruleset
```
